src/lib/find-files.ts

```diff
@@ -102,20 +123,30 @@ async function findInDirectory(
   ignore: string[] = [],
   filter: string[] = [],
   levelsDeep = 4,
-): Promise<string[]> {
+): Promise<FindFilesRes> {
   const files = await readDirectory(path);
   const toFind = files
     .filter((file) => !ignore.includes(file))
     .map((file) => {
       const resolvedPath = pathLib.resolve(path, file);
       if (!fs.existsSync(resolvedPath)) {
         debug('File does not seem to exist, skipping: ', file);
-        return [];
+        return { files: [], allFilesFound: [] };
       }
       return find(resolvedPath, ignore, filter, levelsDeep);
     });
+
   const found = await Promise.all(toFind);
-  return Array.prototype.concat.apply([], found);
+  return {
+    files: Array.prototype.concat.apply(
```

Comment:
using `apply` in combination w/ `map` will make this call allocate 3 arrays; imo we can simply return the mapped array:
```suggestion
    files: found.map(...)
```


> I'm not sure why this specific trick was previously used, as the `this` arg for `concat.apply` is an empty array so the "append" magic `concat.apply` can do is redundant;
 [`concat.apply` example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply#Using_apply_to_append_an_array_to_another)

Reply:
The result is nested arrays, `concat.apply` flattens them all into 1 array.
<img width="747" alt="Screen Shot 2020-08-26 at 17 14 08" src="https://user-images.githubusercontent.com/2911613/91329008-8edf7380-e7bf-11ea-8958-5865a632852c.png">

Reply:
I think I can use `.flat()` will try it!

Reply:
@aviadatsnyk .flat() is not Node 8 compatible and couldn't get it working with TS with es2019 either :/ so reverted back for now until we drop Node 8